include/openPMD/RecordComponent.hpp

```diff
@@ -225,29 +197,118 @@ class RecordComponent : public BaseRecordComponent
     template <typename T>
     std::shared_ptr<T> loadChunk(Offset = {0u}, Extent = {-1u});
 
-    /** Load a chunk of data into pre-allocated memory
+    /** Load a chunk of data into pre-allocated memory.
      *
-     * shared_ptr for data must be pre-allocated, contiguous and large enough
-     * for extent
+     * @param data   Preallocated, contiguous buffer, large enough to load the
+     *               the requested data into it.
+     *               The shared pointer must either own and manage the buffer
+     *               or have been created via shareRaw().
+     *               If using shareRaw(), it is in the user of this API call's
+     *               responsibility to ensure that the lifetime of the buffer
+     *               exceeds the next <a
+     * href="https://openpmd-api.readthedocs.io/en/latest/usage/workflow.html#deferred-data-api-contract">
+     *               flush point</a>.
+     *               Optimizations might be implemented based on this
+     *               assumption (e.g. skipping the operation if the backend
+     *               is the unique owner).
+     *               For raw pointers, use loadChunkRaw().
+     * @param offset Offset within the dataset. Set to {0u} for full selection.
+     * @param extent Extent within the dataset, counted from the offset.
+     *               Set to {-1u} for full selection.
+     *               If offset is non-zero and extent is {-1u} the leftover
+     *               extent in the record component will be selected.
+     */
+    template <typename T>
+    void loadChunk(std::shared_ptr<T> data, Offset offset, Extent extent);
```

Comment:
Do we need a `std::shared_ptr<T[]>`, too?

Reply:
Yeah, I forgot that in #1296. Should I open a separate PR or just add it into this one?